unicef/juniper-portfolio
# Docker files and necessary changes

Ok this PR merges the REST and Graphql apps. 

@mehranhydary 
 Express will accept another instance of express with ```.use```

https://github.com/unicef/gilgamesh-track/compare/master...tenthirtyone:dockerize?expand=1#diff-ed009b6b86b017532ef0489c77de5100R43

It is possible to break the entire app into parts - (Accounts, Wallets, Price Tracker) with their own express instances and pass them up into a parent app. Not useful now but maybe on future projects for breaking out work. 

Changes:
1) Dockerfile
2) Standardize the ENV vars
3) Serve the built front end apps

To Test:
1) fetch the branch
2) ```npm install``` in the app root
[This line](https://github.com/unicef/gilgamesh-track/compare/master...tenthirtyone:dockerize?expand=1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R46) Will automatically install the clients and build them
3) ```npm run start```
4) Browse to ```localhost:9000``` for the visual client
5) Browser to ```localhost:9000/admin``` for the admin

Callouts: Some of the images in the visual client do not load. It's because of the way some are referenced and how static files are served by the back end vs webpack in development. I can run through and update these I just want to test the back end and see if I can serve them up without making a change.

ToDo: The admin routes need ```/admin``` prepended on the admin client to keep the browser address consistent. 

This is currently running in Rancher in the ```juniper-staging``` namespace. I goofed when I said Rancher will handle our DNS changes. I found your email to Paramjit for atrium so I sent him a request for juniper and juniper-staging.
